### PR TITLE
Abstract UI components and add tests

### LINKED ISSUE #105: (edited) Publish changes compatible with downstream React wrapper
Hi,

Terrific mapping library guys.  I was wondering if the maintainers and contributors would mind if I publish a react wrapper for this.  Here is the repo:  https://github.com/ChrisLowe-Takor/react-leaflet-distortable-imageoverlay

It's not strictly a wrapper as I had to chop and change quiet a bit to get all the functionality online (had some major problems with the Toolbar and Translating for one thing).  It uses a combination of my own implementations, Leaflet.Path.Transform and modified copy of the Leaflet.DistortableImage distribution.

Would appreciate your permission and will be happy to chip in with some contributions from time to time.

### LINKED ISSUE #112: Create additional separate Scale and Rotate tools for use in downstream UIs
Projects described in #105 need separate tools for scale and rotate. We can copy out sections of the combined Scale/Rotate to make independent parts. 

rotation/scale were [split into separate actions in this diff](https://gist.github.com/jywarren/59f0cb0baec03bee99540ac2af2ecf70/revisions?diff=split#diff-d592453497ece2dffbf1a19cb06db126R669) from @ChrisLowe-Takor (Also [see this line](https://gist.github.com/jywarren/59f0cb0baec03bee99540ac2af2ecf70/revisions?diff=split#diff-d592453497ece2dffbf1a19cb06db126R259)

Here is the full diff; it's not too complex: https://gist.github.com/jywarren/59f0cb0baec03bee99540ac2af2ecf70/revisions?diff=split#diff-d592453497ece2dffbf1a19cb06db126

There's new code for a "scale" handle here:

https://gist.github.com/jywarren/59f0cb0baec03bee99540ac2af2ecf70/revisions?diff=split#diff-d592453497ece2dffbf1a19cb06db126R308

We don't need to include these new tools in the Toolbar because @ChrisLowe-Takor doesn't use the Toolbar in his implementation. So we just need a new handle in a file here, called `ScaleHandle.js`:

https://github.com/publiclab/Leaflet.DistortableImage/tree/master/src/edit

And we should rename `RotateHandle.js` to `RotateAndScaleHandle.js` and preserve the original version there too. Then we can hook up our new handles like is done here:

https://github.com/publiclab/Leaflet.DistortableImage/blob/dc9e64cc084b67bc9ce9c9b9fa72fefe30beac00/src/edit/DistortableImage.Edit.js#L42

Later we could make things less redundant by sharing some utility functions. But let's not worry about that now.

So to be clear, we'll end with three handles:

1. the original `RotateHandle.js` renamed to `RotateAndScaleHandle.js`
2. the new `RotateHandle.js` which ONLY rotates (as per the diff-modified version above)
3. the new `ScaleHandle.js` which ONLY scales (as per the diff-modified version above)

And matching code in `/src/edit/DistortableImage.Edit.js` to connect them all up. 

The Toolbar code will probably need to be modified to link only to `RotateAndScaleHandle`

Finally, this will take some chasing around the codebase to ensure the new files are included in Grunt, and all references to `RotateHandle` are updated to `RotateAndScaleHandle`:

https://github.com/publiclab/Leaflet.DistortableImage/search?q=rotate&unscoped_q=rotate

This is a bit more complex, and tests will have to be updated, but we can do this step by step and I'm happy to help out if anyone wants to take on this project!

### PR DESCRIPTION
**Pull request by rexagod**
Fixes #105, #112. The sequence of keypress in preview below was "CAPS" first and then the "s" key, *nothing else was clicked or pressed*.

peek 2019-02-19 00-00

- Abstracted UI components (**Rotate and Scale, individual components**) and add tests for the same (also for the originally renamed `RotateAndScaleHandle.js` at `test/src/edit/RotateAndScaleHandleSpec.js`).
- The standalone `rotate` and `scale` functionalities are now mapped to "CAPS" and "s" key respectively, for better user accessibility, but can be removed (I originally did this to test these features out and decided to go along with it).
- Also I didn't include these seperately in the toolbar as per @jywarren's suggestions.

### CODE REVIEW

**Comment by rexagod** [comment]
#140

**Comment by jywarren** [comment]
Ooh wow! Will review today!

File: dist/leaflet.distortableimage.js [hunk 1]
```diff
@@ -303,6 +303,132 @@ L.RotateHandle = L.EditHandle.extend({
...
...MxMjA5IHYtODYuOTI2MjM0NTEyNTUwMDIgaC0yMy44MDk2MDM3MTAwODY2OSBWODYuOTI2MjM0NTEyNTUwMDIgSDQ1OS4wNDkxNTk1MzA0NzEzNCB6TTM4NC43NjMxOTU5NTUwMDA5LDEyLjU1NjAxMTY1MTgxMjc4MSBoNjEuOTA0OTY5NjQ2MjI1Mzk2IHY2Mi43ODAwNTgyNTkwNjM5MSBoLTYxLjkwNDk2OTY0NjIyNTM5NiBWMTIuNTU2MDExNjUxODEyNzgxIHpNMTIuMzgwOTkzOTI5MjQ1MDUsMTIuNTU2MDExNjUxODEyNzgxIGg2MS45MDQ5Njk2NDYyMjUzOTYgdjYyLjc4MDA1ODI1OTA2MzkxIEgxMi4zODA5OTM5MjkyNDUwNSBWMTIuNTU2MDExNjUxODEyNzgxIHpNNzQuMjg1OTYzNTc1NDcwNTMsNDUxLjA1MDU3MjQxNTEyMDY2IEgxMi4zODA5OTM5MjkyNDUwNSB2LTYyLjc4MDA1ODI1OTA2MzkxIGg2MS45MDQ5Njk2NDYyMjUzOTYgVjQ1MS4wNTA1NzI0MTUxMjA2NiB6TTQ0NS43MTU3ODE0NTI4MjI3NCw0NTEuMDUwNTcyNDE1MTIwNjYgaC02Mi44NTczNTM3OTQ2Mjg4NjQgdi02Mi43ODAwNTgyNTkwNjM5MSBoNjIuODU3MzUzNzk0NjI4ODY0IFY0NTEuMDUwNTcyNDE1MTIwNjYgek00MDcuNjIwNDE1NTE2Njg0MjYsMzc2LjY4MDM0OTU1NDM4MzQ0IGgtMzYuMTkwNTk3NjM5MzMxNzcgdjMyLjgzODc5OTcwNDc0MTEyIEg4NS43MTQ1NzMzNTYzMTIwMyB2LTMyLjgzODc5OTcwNDc0MTEyIEg0OS41MjM5NzU3MTY5ODAzMiBWODYuOTI2MjM0NTEyNTUwMDIgaDM2LjE5MDU5NzYzOTMzMTc3IFY1MC4yMjQwNDY2MDcyNTExMjUgaDI4Ny42MjAwMTI4MTc4NDcyIHYzNi43MDIxODc5MDUyOTg5IGgzNC4yODU4MjkzNDI1MjQ4MzUgVjM3Ni42ODAzNDk1NTQzODM0NCB6IiBpZD0ic3ZnXzIiIGNsYXNzPSIiIGZpbGw9IiMxYTFhZWIiIGZpbGwtb3BhY2l0eT0iMSIvPjwvZz48L3N2Zz4=',
```
Comment (jywarren) [inline dist/leaflet.distortableimage.js]: This is really a fun handle!

image

However, what about one that more specifically shows a scale icon, like:

image

Anyways I am happy to merge this and we can make that change later if we wish to.

**Comment by jywarren** [comment]
OK, my apologies I didn't get to review this, as we had the Barnraising event. I just want to say you've been doing some epic work on this project @rexagod !!! Super impressive!

**Comment by jywarren** [comment]
Ugh, sorry there was a conflict in /dist/... so I used the GitHub interface to try to resolve it. I'm going to try to do a clean rebase now.

**Comment by jywarren** [comment]
Publishing as `v0.3.0`!

**Pull request merged**